NuMPI/Optimization/generic_cg_polonsky.py

```diff
@@ -0,0 +1,429 @@
+import numpy as np
+import scipy.optimize as optim
+from inspect import signature
+
+
+def min_cg(objective, hessp, x0=None, gtol=1e-8, mean_value=None,
+           bugnicourt=False, polonskykeer=False,
```

Comment:
What is the difference between setting `bugnicourt=True` here and calling the dedicated `bugnicourt` implementation?